### PR TITLE
Change payload source ready state

### PR DESCRIPTION
**Pull request by jkonecny12**
Payload ready state is not working for sources like URL which does not have a state. Problem with this is that Payload is not able to switch sources which are ready and on the other hand can't start installation with sources which are not ready.

Make this a 3 state - ready, unready and not implemented. That should be enough to solve any problem.

### CODE REVIEW

**Comment by poncovka** [comment]
I think this PR should be rebased on a top of https://github.com/rhinstaller/anaconda/pull/2442.

**Comment by jkonecny12** [comment]
**UPDATED:**
* Rebased and fixed conflicts

File: pyanaconda/modules/payloads/constants.py [hunk 1]
```diff
@@ -48,3 +48,28 @@ class URLType(Enum):
...
+    READY = auto()
+    UNREADY = auto()
+    NOT_SUPPORTED = auto()
+
+    @staticmethod
+    def from_bool(value):
+        """Get state from a bool value.
+
+        This way we can't return NONE state but that is not problem. NONE state is specific
+        for sources which does not have a state, so they don't have to convert it from bool.
```
Comment (M4rtinK) [inline pyanaconda/modules/payloads/constants.py:68]: not problem -> not a problem

does not have -> do not have

**Comment by jkonecny12** [comment]
**UPDATED:**
* Fixed issue pointed by @M4rtinK

**Comment by jkonecny12** [comment]
**UPDATED:**
* Renamed `NOT_SUPPORTED` state to `NOT_APPLICABLE`

**Pull request merged**